src/rpc/blockchain.cpp

```diff
@@ -1363,26 +1367,30 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
 
     LOCK(cs_main);
 
-    double nDiffPoW = 0;
+    double nDiffX16rt = 0;
+    double nDiffRandomX = 0;
+    double nDiffProgPow = 0;
+    double nDiffSha256D = 0;
     double nDiffPoS = 0;
-    auto pindex = chainActive.Tip();
-    if (pindex->IsProofOfWork()) {
-        nDiffPoW = GetDifficulty(pindex);
-        while (pindex->pprev) {
-            pindex = pindex->pprev;
-            if (pindex->IsProofOfStake()) {
-                nDiffPoS = GetDifficulty(pindex);
-                break;
-            }
+    int nBlockCount = 0;
+    
+    // Get the last diff for each algo type
+    CBlockIndex *pindex = chainActive.Tip();
+    while (pindex->pprev) {
+        nBlockCount++;
+        if (pindex->IsProofOfStake() && nDiffPoS == 0) {
+            nDiffPoS = GetDifficulty(pindex);
+        } else if (pindex->IsProgProofOfWork() && nDiffProgPow == 0) {
+            nDiffProgPow = GetDifficulty(pindex);
+        } else if (pindex->IsRandomXProofOfWork() && nDiffRandomX == 0) {
+            nDiffRandomX = GetDifficulty(pindex);
+        } else if (pindex->IsSha256DProofOfWork() && nDiffSha256D == 0) {
+            nDiffSha256D = GetDifficulty(pindex);
+        } else if (pindex->IsX16RTProofOfWork() && nDiffX16rt == 0) {
+            nDiffX16rt = GetDifficulty(pindex);
         }
-    } else {
-        nDiffPoS = GetDifficulty(pindex);
-        while (pindex->pprev) {
-            pindex = pindex->pprev;
```

Comment:
@codeofalltrades ; where does pindex iterate down the chain?  I think you're missing the `pindex= pindex->pprev`; at the bottom of the loop.

I'm guessing the while loop runs the full 1440 before breaking out and you only end up with the difficulty for the chain tip.